File: lcof/go/qing-wa-tiao-tai-jie-wen-ti-lcof.go
/*
 * @Author: FisherCloud 
 * @Date: 2020-02-28 20:14:15 
 * @Last Modified by: FisherCloud
 * @Last Modified time: 2020-02-28 20:16:33
 */
//  https://leetcode-cn.com/problems/qing-wa-tiao-tai-jie-wen-ti-lcof/

package main

func numWays(n int) int {
	var dp []int = []int{1, 1}
	for i := 2; i <= n; i++ {
		dp = append(dp, (dp[i - 1] + dp[i - 2]) % 1000000007);
	}
	return dp[n];
}

File: lcof/rs/qing-wa-tiao-tai-jie-wen-ti-lcof.rs
/*
 * @Author: FisherCloud 
 * @Date: 2020-02-28 20:02:23 
 * @Last Modified by: FisherCloud
 * @Last Modified time: 2020-02-28 20:07:16
 */
// https://leetcode-cn.com/problems/qing-wa-tiao-tai-jie-wen-ti-lcof/

#[derive(Debug)]
struct Solution {
}

impl Solution {
    pub fn num_ways(n: i32) -> i32 {
        let mut dp = vec![1, 1];
        let len = n as usize;
        for i in 2..(len + 1) {
            dp.push((dp[i - 1] + dp[i - 2]) % 1000000007);
        }
        return dp[len];
    }
}
